platform/features/layout/src/LayoutController.js

```diff
@@ -296,6 +310,98 @@ define(
             this.endDragInScope();
         };
 
+        /**
+         * Check if the object is currently selected, or (if no
+         * argument is supplied) get the currently selected object.
+         *
+         * @param {string} obj the object to select
+         * @returns {boolean} true if selected
+         */
+        LayoutController.prototype.selected = function (obj) { 
+            var selection = this.selection;
+            return selection && ((arguments.length > 0) ?
+                    selection.selected(obj) : selection.get());
+        };
+
+        /**
+         * Set the active user selection in this view.
+         *
+         * @param event the mouse event
+         * @param {string} obj the object to select
+         */
+        LayoutController.prototype.select = function select(event, obj) {
+            event.stopPropagation();
+
+            var self = this;
+            
+            // Toggle the visibility of the object frame
+            function toggle() {                                
+                var id = obj.getId();
+                var configuration = self.$scope.configuration;
+
+                configuration.panels[id].hasFrame = 
+                    !configuration.panels[id].hasFrame;
+
+                // Change which method is exposed, to influence
+                // which button is shown in the toolbar
+                delete obj[SHOW];
+                delete obj[HIDE];
+                obj[configuration.panels[id].hasFrame ? HIDE : SHOW] = toggle;
+
+                var selection = self.selection;
+
+                obj.useCapability('mutation', function() {
```

Comment:
Here you're mutating the object, what you should be doing is listening for mutations and then deselecting / reselecting the object.  you need to attach it as a callback, e.g. `obj.getCapability('mutation').listen(function() {...});`